Add support for nightly and rc versions (#611)
actions/setup-node: docs/advanced-usage.md

      - run: npm test
```

## Nightly versions

You can specify a nightly version to download it from https://nodejs.org/download/nightly.

```yaml
jobs:
  build:
    runs-on: ubuntu-latest
    name: Node sample
    steps:
      - uses: actions/checkout@v3
      - uses: actions/setup-node@v3
        with:

>           node-version: '16.0.0-nightly' # or 16-nightly

Reviewer: To avoid ambiguity, I think it's worth a separate explanation of what the `16-nightly` syntax means.  From my understanding, it means the same thing as saying `node-version: 16` but for nightlies.  So, it will find the nightly for the latest Node 16 (as opposed to 16.0.0).

## Nightly versions

You can specify a nightly version to download it from https://nodejs.org/download/nightly. 

> ### Install nightly build for specific node version

Reviewer: ```suggestion
### Install the nightly build for a specific Node version
```

```yaml
jobs:
  build:
    runs-on: ubuntu-latest
    name: Node sample
    steps:
      - uses: actions/checkout@v3
      - uses: actions/setup-node@v3
        with:

>           node-version: '16.0.0-nightly' # it will install the latest nigthly release for node 16.0.0

Reviewer: ```suggestion
          node-version: '16.0.0-nightly' # it will install the latest nightly release for node 16.0.0
```

      - run: npm ci
      - run: npm test
```

> ### Install nightly build for major node version

Reviewer: ```suggestion
### Install the nightly build for a major Node version
```

Reviewer: Also, just for a better flow, I would move this first so it goes major version -> specific version -> exact nightly

      - uses: actions/checkout@v3
      - uses: actions/setup-node@v3
        with:

>           node-version: '16-nightly' # it will install the latest nigthly release for node 16

Reviewer: ```suggestion
          node-version: '16-nightly' # it will install the latest nightly release for node 16
```

      - run: npm ci
      - run: npm test
```

> ### Install the exact nightly version

Reviewer: ```suggestion
### Install an exact nightly version
```

      - run: npm test
```

> **Note:** You should specify the exact version for rc: `16.0.0-rc.1`.

Reviewer: ```suggestion
**Note:** Unlike nightly versions, which support version range specifiers, you must specify the exact version for a release candidate: `16.0.0-rc.1`.
```
